bindings/cs/MongoDB.Libmongocrypt/LibraryLoader.cs

```diff
@@ -58,42 +58,46 @@ public LibraryLoader()
-//            Trace.WriteLine("Base Path: " + basepath)
 
-#if NETSTANDARD1_5
-            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            switch (OperationSystemHelper.CurrentOperationSystem)
             {
-                string[] suffixPaths = new[]
-                {
-                    "../../native/osx/",
-                    string.Empty
-                };
-                string path = FindLibrary(candidatePaths, suffixPaths, "libmongocrypt.dylib");
-                _loader = new DarwinLibraryLoader(path);
-            }
-            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
-            {
-                string[] suffixPaths = new[]
-                {
-                    "../../native/linux/",
-                    string.Empty
-                };
-                string path = FindLibrary(candidatePaths, suffixPaths, "libmongocrypt.so");
-                _loader = new LinuxLibrary(path);
-            }
-            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
-
-#endif
-            {
-                string[] suffixPaths = new[]
-                {
-                    @"..\..\x64\native\windows\",
-                    string.Empty
-                };
-                string path = FindLibrary(candidatePaths, suffixPaths, "mongocrypt.dll");
-                _loader = new WindowsLibrary(path);
+                case OperationSystemPlatform.MacOS:
+                    {
+                        string[] suffixPaths = new[]
+                        {
+                            "../../native/osx/",
+                            string.Empty
+                        };
+                        string path = FindLibrary(candidatePaths, suffixPaths, "libmongocrypt.dylib");
+                        _loader = new DarwinLibraryLoader(path);
+                    }
+                    break;
+                case OperationSystemPlatform.Linux:
+                    {
+                        string[] suffixPaths = new[]
+                        {
+                            "../../native/linux/",
+                            string.Empty
+                        };
+                        string path = FindLibrary(candidatePaths, suffixPaths, "libmongocrypt.so");
+                        _loader = new LinuxLibrary(path);
+                    }
+                    break;
+                case OperationSystemPlatform.Windows:
+                    {
+                        string[] suffixPaths = new[]
+                        {
+                            @"..\..\x64\native\windows\",
+                            string.Empty
+                        };
+                        string path = FindLibrary(candidatePaths, suffixPaths, "mongocrypt.dll");
+                        _loader = new WindowsLibrary(path);
+                    }
+                    break;
+                default:
+                    // should not be reached. If we're here, then there is a bug in OperationSystemHelper
```

Comment:
As far as I can tell you didn't need to make this change. Looks like you are only refactoring from if statements to a switch statement?

Reply:
Oh... I guess the default clause is new?

If you had simply added a final else the change would have been easier to understand...

Reply:
From my point of view, `switch` is more good-looking here. As I wrote in the first comment, `default` is actually can be merged with `windows`, in other words, `default` can contains `windows` logic, but I think it's better to mention all supported platforms explicitly. So I would live it as is

Reply:
Totally agree that switch is good. Just pointing out that making somewhat unnecessary changes complicates the code review. I had to look hard at code that didn't really change to figure out why this change was made.

Reply:
My fault, I needed to explain it more deeply. I made this change because the same condition was used in another code part. But we temporally removed it since we noticed that in the previous release we missed some linux/macos related method calls. The missed methods will be added in the scope of linux/macos supporting. So, I decided that it's harmless to leave this code in a new form